### PR TITLE
chore(flutter-docs): use latest version of flutter and amplify, miscellaneous updates

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
#### Issue #, if available

Original Issue: https://github.com/aws-amplify/amplify-flutter/issues/3493

That issue was the original reason for the PR, but this PR also makes other updates to bring the docs up to date with the latest version of flutter, the amplify authenticator, and other packages.

#### Description of how you validated changes

Deployed to https://chore-update-flutter-routing-docs.d1m6l3c1y80m52.amplifyapp.com/flutter/connected-components/authenticator

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/theming/theming.flutter.mdx [hunk 1]
```diff
@@ -1,4 +1,8 @@
-import { Link } from '@aws-amplify/ui-react';
+import { Alert, Link } from '@aws-amplify/ui-react';
+
+<Alert variation="info" role="none">
+  All of the live examples in the docs opt into material 3 support. This will become the default in a future version Flutter. See <Link isExternal href="https://api.flutter.dev/flutter/material/ThemeData/useMaterial3.html">useMaterial3</Link> for more info.
```
Comment (calebpollman) [inline docs/src/pages/[platform]/theming/theming.flutter.mdx:4]:
```suggestion
The live examples opt into usage of Material 3 which will become the default in a future version of Flutter. See <Link isExternal href="https://api.flutter.dev/flutter/material/ThemeData/useMaterial3.html">useMaterial3</Link> for more info.

**Review by calebpollman — COMMENTED** [review]
LGTM, just left a suggestion

**Review by Equartey — DISMISSED** [review]
LGTM

File: docs/src/pages/[platform]/connected-components/authenticator/customization/customization.i18n.flutter.mdx [hunk 2]
```diff
@@ -54,8 +54,8 @@ Finally, update your MaterialApp:
 ```dart
 return MaterialApp(
   title: 'Authenticator Demo',
-  theme: ThemeData.light(),
-  darkTheme: ThemeData.dark(),
+  theme: ThemeData.light(useMaterial3: true,),
+  darkTheme: ThemeData.dark(useMaterial3: true,),
```
Comment (dnys1) [inline docs/src/pages/[platform]/connected-components/authenticator/customization/customization.i18n.flutter.mdx:58]:
```suggestion
  theme: ThemeData.light(useMaterial3: true),
  darkTheme: ThemeData.dark(useMaterial3: true),
```

File: docs/src/pages/[platform]/connected-components/authenticator/customization/customization.labels-and-text.flutter.mdx [hunk 3]
```diff
@@ -51,8 +51,8 @@ final authenticator = Authenticator(
   stringResolver: stringResolver,
   child: MaterialApp(
     title: 'Authenticator Demo',
-    theme: ThemeData.light(),
-    darkTheme: ThemeData.dark(),
+    theme: ThemeData.light(useMaterial3: true,),
+    darkTheme: ThemeData.dark(useMaterial3: true,),
```
Comment (dnys1) [inline docs/src/pages/[platform]/connected-components/authenticator/customization/customization.labels-and-text.flutter.mdx:55]:
```suggestion
    theme: ThemeData.light(useMaterial3: true),
    darkTheme: ThemeData.dark(useMaterial3: true),
```

File: docs/flutter/authenticator/lib/main.dart [hunk 4]
```diff
@@ -112,15 +112,18 @@ class _MyAppState extends State<MyApp> {
...
   }
 
   ThemeData get theme {
-    final theme =
-        widget.config.useCustomTheme ? customLightTheme : ThemeData.light();
+    final theme = widget.config.useCustomTheme
+        ? customLightTheme
+        : ThemeData.light(
+            useMaterial3: true,
+          );
```
Comment (dnys1) [inline docs/flutter/authenticator/lib/main.dart:126]:
```suggestion
        : ThemeData.light(useMaterial3: true);
```

File: docs/flutter/authenticator/lib/main.dart [hunk 5]
```diff
@@ -130,8 +133,11 @@ class _MyAppState extends State<MyApp> {
   }
 
   ThemeData get darkTheme {
-    final theme =
-        widget.config.useCustomTheme ? customDarkTheme : ThemeData.dark();
+    final theme = widget.config.useCustomTheme
+        ? customDarkTheme
+        : ThemeData.dark(
+            useMaterial3: true,
+          );
```
Comment (dnys1) [inline docs/flutter/authenticator/lib/main.dart:140]:
```suggestion
        : ThemeData.dark(useMaterial3: true);
```

File: docs/src/pages/[platform]/theming/theming.flutter.mdx [hunk 6]
```diff
@@ -1,4 +1,8 @@
-import { Link } from '@aws-amplify/ui-react';
+import { Alert, Link } from '@aws-amplify/ui-react';
+
+<Alert variation="info" role="none">
+The live examples opt into usage of Material 3 which will become the default in a future version of Flutter. See <Link isExternal href="https://api.flutter.dev/flutter/material/ThemeData/useMaterial3.html">useMaterial3</Link> for more info.
```
Comment (dnys1) [inline docs/src/pages/[platform]/theming/theming.flutter.mdx:4]:
```suggestion
The live examples use Material 3 styling which is currently behind a feature flag. See <Link isExternal href="https://api.flutter.dev/flutter/material/ThemeData/useMaterial3.html">useMaterial3</Link> for more info. The next stable release of Flutter is expected to make Material 3 the default.
```

File: docs/flutter/authenticator/pubspec.yaml [hunk 7]
```diff
@@ -4,21 +4,23 @@ publish_to: 'none'
...
     sdk: flutter
-  amplify_flutter: ^1.0.0-0
-  amplify_auth_cognito: ^1.0.0-0
-  amplify_authenticator: ^1.0.0-0
-  device_preview: ^1.0.0
-  confetti: ^0.7.0
+  go_router: ^10.0.0
 
 dev_dependencies:
+  amplify_lints: ^2.0.2
```
Comment (dnys1) [inline docs/flutter/authenticator/pubspec.yaml:21]:
```suggestion
  amplify_lints: ^3.0.0
```

**Review by dnys1 — APPROVED** [review]
All nits - looks good

**Review by calebpollman — APPROVED** [review]

**Pull request merged**